src/web/parserUtils.ts

```diff
@@ -179,19 +179,22 @@ function parseText(
 
   const markdownRanges: MarkdownRange[] = ranges as MarkdownRange[];
 
-  targetElement.innerHTML = '';
-  targetElement.innerText = '';
-
   // We don't want to parse text with single '\n', because contentEditable represents it as invisible <br />
   if (!!text && text !== '\n') {
     const dom = parseRangesToHTMLNodes(text, markdownRanges, markdownStyle);
-    target.appendChild(dom);
-  }
 
-  if (alwaysMoveCursorToTheEnd) {
-    CursorUtils.moveCursorToEnd(target);
-  } else if (isFocused && cursorPosition !== null) {
-    CursorUtils.setCursorPosition(target, cursorPosition, disableNewLinesInCursorPositioning);
+    // eslint-disable-next-line es/no-optional-chaining
+    if ((targetElement.firstChild as HTMLElement | null)?.innerHTML !== dom.innerHTML) {
+      targetElement.innerHTML = '';
+      targetElement.innerText = '';
+      target.appendChild(dom);
```

Comment:
How about using `replaceWith` method instead? https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceWith

Reply:
It won't work since there's no root span when the component is rendered for the first time, because of which there's no way to replace it